src/NServiceBus.Persistence.AzureTable/Outbox/OutboxOperations.cs

```diff
@@ -1,40 +1,41 @@
 ﻿namespace NServiceBus.Persistence.AzureTable
 {
-    using Microsoft.Azure.Cosmos.Table;
+    using System.Collections.Generic;
+    using Azure.Data.Tables;
 
     class OutboxStore : Operation
     {
-        public OutboxStore(TableEntityPartitionKey partitionKey, OutboxRecord outboxRow, CloudTable cloudTable) : base(partitionKey)
+        public OutboxStore(TableEntityPartitionKey partitionKey, OutboxRecord outboxRow, TableClient cloudTable) : base(partitionKey)
         {
             this.cloudTable = cloudTable;
             this.outboxRow = outboxRow;
         }
 
-        public override CloudTable Apply(TableBatchOperation transactionalBatch)
+        public override TableClient Apply(List<TableTransactionAction> transactionalBatch)
         {
-            transactionalBatch.Add(TableOperation.Insert(outboxRow));
+            transactionalBatch.Add(new TableTransactionAction(TableTransactionActionType.Add, outboxRow));
             return cloudTable;
         }
 
         readonly OutboxRecord outboxRow;
-        readonly CloudTable cloudTable;
+        readonly TableClient cloudTable;
     }
 
     class OutboxDelete : Operation
     {
-        public OutboxDelete(TableEntityPartitionKey partitionKey, OutboxRecord outboxRow, CloudTable cloudTable) : base(partitionKey)
+        public OutboxDelete(TableEntityPartitionKey partitionKey, OutboxRecord outboxRow, TableClient cloudTable) : base(partitionKey)
         {
             this.cloudTable = cloudTable;
             this.outboxRow = outboxRow;
         }
 
-        public override CloudTable Apply(TableBatchOperation transactionalBatch)
+        public override TableClient Apply(List<TableTransactionAction> transactionalBatch)
         {
-            transactionalBatch.Add(TableOperation.Replace(outboxRow));
+            transactionalBatch.Add(new TableTransactionAction(TableTransactionActionType.UpdateReplace, outboxRow));
```

Comment:
Is my understanding correct that this corresponds to https://learn.microsoft.com/en-us/dotnet/api/azure.data.tables.tableclient.updateentity?view=azure-dotnet#azure-data-tables-tableclient-updateentity-1(-0-azure-etag-azure-data-tables-tableupdatemode-system-threading-cancellationtoken) which automatically will make sure the entity loaded ETAG is considered? It is important to verify this because we need to have the same ETAG handling semantics as before

This comment also applies to the other usages of `TableTransactionAction`